core/http-auth-aws/src/main/java/software/amazon/awssdk/http/auth/aws/AwsV4HttpSigner.java

```diff
@@ -13,81 +13,103 @@
      */
     SignerProperty<Clock> SIGNING_CLOCK =
         SignerProperty.create(Clock.class, "SigningClock");
 
     /**
-     * The name of the header for the checksum.
-     * This property is optional.
+     * A boolean to indicate whether to double url-encode the resource path when constructing the canonical request. This property
+     * defaults to true.
      */
-    SignerProperty<String> CHECKSUM_HEADER_NAME =
-        SignerProperty.create(String.class, "ChecksumHeaderName");
+    SignerProperty<Boolean> DOUBLE_URL_ENCODE =
+        SignerProperty.create(Boolean.class, "DoubleUrlEncode");
 
     /**
-     * The {@link ChecksumAlgorithm} used to compute the checksum.
-     * This property is required *if* a checksum-header name is given.
+     * A boolean to indicate whether the resource path should be "normalized" according to RFC3986 when constructing the canonical
+     * request. This property defaults to true.
      */
-    SignerProperty<ChecksumAlgorithm> CHECKSUM_ALGORITHM =
-        SignerProperty.create(ChecksumAlgorithm.class, "ChecksumAlgorithm");
+    SignerProperty<Boolean> NORMALIZE_PATH =
+        SignerProperty.create(Boolean.class, "NormalizePath");
 
     /**
-     * A boolean to indicate whether to double url-encode the resource path
-     * when constructing the canonical request.
-     * This property defaults to true.
+     * The location where auth-related data is inserted, as a result of signing. This property defaults to HEADER.
      */
-    SignerProperty<Boolean> DOUBLE_URL_ENCODE =
-        SignerProperty.create(Boolean.class, "DoubleUrlEncode");
+    SignerProperty<AuthLocation> AUTH_LOCATION =
+        SignerProperty.create(AuthLocation.class, "AuthLocation");
 
     /**
-     * A boolean to indicate whether the resource path should be "normalized"
-     * according to RFC3986 when constructing the canonical request.
-     * This property defaults to true.
+     * The duration for the request to be valid. This property defaults to the max valid duration (7 days). This is only used in
+     * the case of a pre-signing implementation.
      */
-    SignerProperty<Boolean> NORMALIZE_PATH =
-        SignerProperty.create(Boolean.class, "NormalizePath");
+    SignerProperty<Duration> EXPIRATION_DURATION =
+        SignerProperty.create(Duration.class, "ExpirationDuration");
+
+    /**
+     * Whether to enable chunked encoding or not. This property defaults to false. This is only used in the case of an
+     * implementation that supports chunked-encoding.
+     */
+    SignerProperty<Boolean> CHUNKED_ENCODING =
+        SignerProperty.create(Boolean.class, "ChunkedEncoding");
+
+    /**
+     * Whether to indicate that a payload is signed or not. This property defaults to true. This can be set false to disable
+     * payload signing.
+     */
+    SignerProperty<Boolean> PAYLOAD_SIGNING =
+        SignerProperty.create(Boolean.class, "PayloadSigning");
```

Comment:
Nit: PAYLOAD_SIGNING_ENABLED to match our normal naming convention for boolean setters/getters?

Reply:
Same above for other boolean settings.

Reply:
Just checking convention. Would it be PAYLOAD_SIGNING_ENABLED or ENABLE_PAYLOAD_SIGNING which is what was used in [S3SignerExecutionAttribute](https://github.com/aws/aws-sdk-java-v2/blob/52b158dbe6662fc50249527d4b7a6ebb8e2598d5/core/auth/src/main/java/software/amazon/awssdk/auth/signer/S3SignerExecutionAttribute.java#L36C53-L36C75)